scripts/provisioner.sh

```diff
@@ -47,7 +47,17 @@ do
   fi
 done
 set -e
-sleep 10
+
+# If asked to, wait for the total number of control-plane nodes to be registered.
+set +e
+if [[ "${WAIT_FOR_TOTAL_CONTROL_PLANE_NODES}" == "true" ]];
+then
+  until [[ $(kubectl get node -l node-role.kubernetes.io/control-plane --no-headers | wc -l) == "${TOTAL_CONTROL_PLANE_NODES}" ]];
```

Comment:
In case the label `node-role.kubernetes.io/control-plane` doesn't exist this is going to result in an infinite loop and will never terminate.
To give some examples where this may occur:
- Special configuration where people wanted/had to to change that label and we must test with it (i.e. for migration purposes had to stick with the old `node-role.kubernetes.io/master`)
- Kubernetes decides yet again to change names of well-known labels
- Managed Kubernetes where there is no access to the control-plane

Of course this is not even enabled by default but I'd still advise to use a proper loop instead and iterate over it a maximum of X times before throwing an error.
Otherwise this may lead to head-scratching people waiting forever in certain corner cases.
Perhaps make the label configurable as well.

Reply:
Good shout -- I'll improve upon this in a subsequent PR 💯

Reply:
Opened https://github.com/isovalent/terraform-k8s-cilium/pull/13.